tests/run-make/split-debuginfo/rmake.rs

```diff
@@ -0,0 +1,428 @@
+    match params.split_dwarf {
+        SplitDwarf::Split => compiler.arg("-Zsplit-dwarf-kind=split"),
+        SplitDwarf::Single => compiler.arg("-Zsplit-dwarf-kind=single"),
+    };
+    if let Some(crate_type) = params.crate_type {
+        match crate_type {
+            CrateType::Rlib => compiler.crate_type("rlib"),
+            CrateType::Lib => compiler.crate_type("lib"),
+        };
+    };
+    if params.input == "lto.rs" {
+        compiler.arg("-Clinker-plugin-lto");
+    }
+    if params.input == "main.rs" {
+        compiler.extern_("bar", rust_lib_name("bar"));
+    }
+    if let Some(remap) = params.remap_scope {
+        compiler.remap_path_prefix(cwd(), "/a");
+        if let Some(scope) = remap {
+            match scope {
+                Scope::Debug => compiler.arg("-Zremap-path-scope=debuginfo"),
+                Scope::Macro => compiler.arg("-Zremap-path-scope=macro"),
+            };
+        };
+    }
+    compiler.run();
+    if let Some(remap) = params.remap_scope {
+        let mut objdump = llvm_dwarfdump();
+        let out = objdump.arg("--debug-info").input("foo").run();
+        out.assert_stdout_contains("DW_AT_GNU_dwo_name");
+        if let Some(scope) = remap {
+            match scope {
+                Scope::Debug => out.assert_stdout_not_contains(cwd().display().to_string()),
+                Scope::Macro => out.assert_stdout_contains(cwd().display().to_string()),
+            };
+        } else {
+            out.assert_stdout_not_contains(cwd().display().to_string());
+        };
+    }
+    if params.has_o {
+        for object in shallow_find_files(cwd(), |path| has_extension(path, "o")) {
+            rfs::remove_file(object);
+        }
+    } else {
+        check_extension_not_exists("o");
+    }
+    if params.has_dwo {
+        for dwo in shallow_find_files(cwd(), |path| has_extension(path, "dwo")) {
+            rfs::remove_file(dwo);
+        }
+    } else {
+        check_extension_not_exists("dwo");
+    }
+    if let Some(name) = params.dwp {
+        rfs::remove_file(&format!("{name}.dwp"));
+    } else {
+        check_extension_not_exists("dwp");
+    }
+    if let Some(output) = params.output {
+        rfs::remove_file(output);
+    }
+}
+
+struct DebugInfoTest {
+    input: &'static str,
+    crate_type: Option<CrateType>,
+    debug_info: DebugInfo,
+    split_dwarf: SplitDwarf,
+    has_o: bool,
+    has_dwo: bool,
+    dwp: Option<&'static str>,
+    output: Option<String>,
+    remap_scope: Option<Option<Scope>>,
+}
+
+enum CrateType {
+    Lib,
+    Rlib,
+}
+
+#[derive(Clone, Copy)]
+enum Scope {
+    Debug,
+    Macro,
+}
+
+enum DebugInfo {
+    Packed,
+    Unpacked,
+}
+
+#[derive(Clone, Copy)]
+enum SplitDwarf {
+    Split,
+    Single,
+}
+
+// Some non-Windows, non-Darwin platforms are not stable, and some are.
+fn unstable_rustc() -> Rustc {
+    let mut compiler = rustc();
+    if !target().contains("linux") {
+        compiler.arg("-Zunstable-options");
+    }
```

Comment:
Question: the impl here is for non-linux, it doesn't quite match up with the comment?

Reply:
The comment is from the original Makefile:

```
# Some non-Windows, non-Darwin platforms are not stable, and some are.
ifeq ($(UNAME),Linux)
  UNSTABLEOPTS :=
else
  UNSTABLEOPTS := -Zunstable-options
endif
```

I am open to inverting the comment to read "Some non-Linux platforms are not stable, and some are."